### PR TITLE
Update CHANGELOG and file versioning for upcoming release

### PR DESCRIPTION
**Pull request by aggarw13**
Update **CHANGELOG** and **source file versioning** for upcoming release.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,12 @@
 # Changelog for backoffAlgorithm Library
 
-## v1.0.0 December 2020
+## v1.0.1 (March 2020)
+
+### Changes
+
+- [#18](https://github.com/FreeRTOS/backoffAlgorithm/pull/19) and [#20](https://github.com/FreeRTOS/backoffAlgorithm/pull/20) Documentation fixes.
```
Comment (muneebahmed10) [inline CHANGELOG.md:7]: Text says 18, but URL points to 19

File: test/unit-test/catch_assert.h [hunk 2]
```diff
@@ -62,17 +62,18 @@ static void catchHandler_( int signal )
         close( 2 );                          \
         if( setjmp( CATCH_JMPBUF ) == 0 )    \
         {                                    \
-            try++;                           \
+            try ++;                          \
```
Comment (muneebahmed10) [inline test/unit-test/catch_assert.h:65]: What does the space do?

**Review by muneebahmed10 — CHANGES_REQUESTED** [review]

File: test/unit-test/catch_assert.h (hunk 2 above)
Comment (aggarw13) [reply test/unit-test/catch_assert.h:65]: This is auto-formatted by the VS code extension I use for uncrustify.
Fixed

File: CHANGELOG.md (hunk 1 above)
Comment (aggarw13) [reply CHANGELOG.md]: Both #18 and #19 are needed. Fixed.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,12 @@
 # Changelog for backoffAlgorithm Library
 
-## v1.0.0 December 2020
+## v1.0.1 (March 2020)
+
+### Changes
+
+- [#18](https://github.com/FreeRTOS/backoffAlgorithm/pull/18), [#19](https://github.com/FreeRTOS/backoffAlgorithm/pull/19) and [#20](https://github.com/FreeRTOS/backoffAlgorithm/pull/20) Documentation fixes.
```
Comment (muneebahmed10) [inline CHANGELOG.md:7]: No Oxford comma? Disappointing
```suggestion
- [#18](https://github.com/FreeRTOS/backoffAlgorithm/pull/18), [#19](https://github.com/FreeRTOS/backoffAlgorithm/pull/19), and [#20](https://github.com/FreeRTOS/backoffAlgorithm/pull/20) Documentation fixes.
```

**Review by muneebahmed10 — APPROVED** [review]

**Review by muneebahmed10 — APPROVED** [review]

File: test/unit-test/backoff_algorithm_utest.c [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /*
- * backoffAlgorithm v1.0.0
+ * backoffAlgorithm V1.0.1
```
Comment (muneebahmed10) [inline test/unit-test/backoff_algorithm_utest.c:2]: Actually wait why are these ones uppercase

Comment (aggarw13) [reply test/unit-test/backoff_algorithm_utest.c]: Accidental Search + Replace. Fixed

**Review by muneebahmed10 — APPROVED** [review]

**Review by muneebahmed10 — APPROVED** [review]

**Review by aggarg — APPROVED** [review]

**Pull request merged**